src/Network/NWEndpoint.cs

```diff
@@ -0,0 +1,163 @@
+//
+// NWEndpoint.cs: Bindings the Netowrk nw_endpoint_t API.
+//
+// Authors:
+//   Miguel de Icaza (miguel@microsoft.com)
+//
+// Copyrigh 2018 Microsoft Inc
+//
+using System;
+using System.Runtime.InteropServices;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+
+using OS_nw_endpoint=System.IntPtr;
+
+namespace Network {
+
+	public enum NWEndpointType {
+		Invalid = 0,
+		Address = 1,
+		Host = 2,
+		BonjourService = 3
+	}
+	
+	public class NWEndpoint : INativeObject, IDisposable {
+		IntPtr handle;
+		public IntPtr Handle {
+			get { return handle; }
+		}
+
+		public NWEndpoint (IntPtr handle, bool owns)
+		{
+			this.handle = handle;
+			if (owns == false)
+				CFObject.CFRetain (handle);
+		}
+
+		public NWEndpoint (IntPtr handle) : this (handle, false)
+		{
+		}
+
+		~NWEndpoint ()
+		{
+			Dispose (false);
+		}
+
+		public void Dispose ()
+		{
+			Dispose (true);
+			GC.SuppressFinalize (this);
+		}
+
+		public virtual void Dispose (bool disposing)
+		{
+			if (handle != IntPtr.Zero) {
+				CFObject.CFRelease (handle);
+				handle = IntPtr.Zero;
+			}
+		}
+
+		[Mac(10,14)][iOS(12,0)][Watch(5,0)][TV(12,0)]
```

Comment:
Sadly the framework is not available on watchOS so all `[Watch(5,0)]` should be removed.

Reply:
It is indeed, the header files show that everywhere :-(